src/System.Runtime/tests/System/GCTests.cs

```diff
@@ -157,14 +169,19 @@ public static void KeepAlive_Null()
 
         private class KeepAliveNullTest
         {
-            public static void Run()
+            [System.Runtime.CompilerServices.MethodImplAttribute(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
+            private static void MakeAndNull()
             {
                 var obj = new TestObject();
                 obj = null;
+            }
+
+            public static void Run()
+            {
+                MakeAndNull();
 
                 GC.Collect();
                 GC.WaitForPendingFinalizers();
-                GC.KeepAlive(obj);
 
```

Comment:
Likewise, I don't understand all of the things that the JIT does, but my understanding is that (before this change), there's no reason that the JIT can't remove the null write to obj on line 163, since it's dead without the `KeepAlive`. At least in this case the JIT is forced to extend the lifetime of obj to the `KeepAlive` call, I guess?